core-backend/wasmtime/src/env.rs

```diff
@@ -31,12 +31,57 @@ use wasmtime::{
     Engine, Extern, Func, Instance, Memory as WasmtimeMemory, MemoryType, Module, Store, Trap,
 };
 
+/// Complitelly same as LaterExt, but with Sync + Send implementations,
+/// which is needed only for wasmtime restrictions and never used actually.
+pub struct SyncLaterExt<E: Ext>(LaterExt<E>);
+
+impl<E: Ext> Default for SyncLaterExt<E> {
+    fn default() -> Self {
+        Self(Default::default())
+    }
+}
+
+impl<E: Ext> Clone for SyncLaterExt<E> {
+    fn clone(&self) -> Self {
+        Self(self.0.clone())
+    }
+}
+
+impl<E: Ext> SyncLaterExt<E> {
```

Comment:
I would only add inner field accessor.
Deref and DerefMut seems the best choice for me here